### PR TITLE
add `any_value` function

### PR DESCRIPTION
**Pull request by Gabriel39**
Issue: [Add a new ANY_VALUE function](https://github.com/StarRocks/starrocks/issues/1938)

`ANY_VALUE` will fail at analyze phase when FULL_GROUP_BY is set to true (default).

Besides, `ANY_VALUE` will choose the first value as the result for performance consideration.

### CODE REVIEW

File: gensrc/thrift/PlanNodes.thrift [hunk 1]
```diff
@@ -406,6 +406,7 @@ struct TMergeJoinNode {
 }
 
 enum TAggregationOp {
+  ANY_VALUE,
```
Comment (imay) [inline gensrc/thrift/PlanNodes.thrift:409]: The new addition needs to be put at the end so that forward compatibility can be achieved.

**Review by imay — COMMENTED** [review]
1. Run ./build-support/clang-format.sh in order to make CPP files comply with code format.
2. Better to add some UT.

File: be/src/exprs/agg/any_value.h [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+    void reset(FunctionContext* ctx, const Columns& args, AggDataPtr state) const override {
+        this->data(state).reset();
+    }
+
+    void update(FunctionContext* ctx, const Column** columns, AggDataPtr __restrict state,
+                size_t row_num) const override {
+        DCHECK(!columns[0]->is_nullable() && !columns[0]->is_binary());
+        const auto& column = down_cast<const InputColumnType&>(*columns[0]);
+        T value = column.get_data()[row_num];
```
Comment (stdpain) [inline be/src/exprs/agg/any_value.h:138]:
```suggestion
        const T& value = column.get_data()[row_num];
```

**Comment by Gabriel39** [comment]
Hi @Seaven @stdpain , thanks for your comments. I checked the usage for both two `update_batch_single_state` function just now and found that one is used for window function and another is used for aggregate function. Since `any_value` is only used as an aggregate function, my question is whether it is enough to implement `virtual void update_batch_single_state(FunctionContext* ctx, size_t batch_size, const Column** columns, AggDataPtr __restrict state) const = 0;` only?

**Comment by Seaven** [comment]
> Hi @Seaven @stdpain , thanks for your comments. I checked the usage for both two `update_batch_single_state` function just now and found that one is used for window function and another is used for aggregate function. Since `any_value` is only used as an aggregate function, my question is whether it is enough to implement `virtual void update_batch_single_state(FunctionContext* ctx, size_t batch_size, const Column** columns, AggDataPtr __restrict state) const = 0;` only?
> 
> Btw, it seems like all uts for be were passed but jenkins script ended with a failure. Something wrong with be uts? http://39.101.206.100:8090/job/starrocks_be_unittest/11175/console

No, should implement `update` method, you can find other method has call `udpate` in `aggregate.h`.

**Comment by Seaven** [comment]
run starrocks_be_unittest

**Comment by Gabriel39** [comment]
> > Hi @Seaven @stdpain , thanks for your comments. I checked the usage for both two `update_batch_single_state` function just now and found that one is used for window function and another is used for aggregate function. Since `any_value` is only used as an aggregate function, my question is whether it is enough to implement `virtual void update_batch_single_state(FunctionContext* ctx, size_t batch_size, const Column** columns, AggDataPtr __restrict state) const = 0;` only?
> > Btw, it seems like all uts for be were passed but jenkins script ended with a failure. Something wrong with be uts? http://39.101.206.100:8090/job/starrocks_be_unittest/11175/console
> 
> No, should implement `update` method, you can find other method has call `udpate` in `aggregate.h`.

@Seaven Yep, I think `update` is necessary in this function.
I mean `virtual void update_batch_single_state(FunctionContext* ctx, size_t batch_size, const Column** columns, AggDataPtr __restrict state) const = 0;` should be implemented while `virtual void update_batch_single_state(FunctionContext* ctx, AggDataPtr __restrict state, const Column** columns, int64_t peer_group_start, int64_t peer_group_end, int64_t frame_start, int64_t frame_end)` is unnecessary because `any_value` is only used for aggregation.

Btw, BE uts failed again.

Should we talk more about this PR in Slack channel? My Slack name is Gabriel Lee and feel free to contact me

**Comment by stdpain** [comment]
@Gabriel39 you could rebase main and run be UT

**Comment by Gabriel39** [comment]
> @Gabriel39 you could rebase main and run be UT

@stdpain It works! Thanks

**Review by stdpain — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**